packages/vaadin-crud/src/vaadin-dialog-layout.js

```diff
@@ -161,33 +166,42 @@ class DialogLayout extends ElementMixin(ThemableMixin(PolymerElement)) {
     this._ensureChildren();
   }
 
+  __moveChildNodes(target) {
+    // Teleport header node
+    target.appendChild(this.__header);
+
+    // For custom form, remove slot attribute
+    // so that it is properly moved to dialog.
+    if (this.form.hasAttribute('slot')) {
+      this.form.removeAttribute('slot');
+    }
```

Comment:
Here we make sure the custom form gets moved in the same way as the default one.